webapp/apps/taxbrain/views.py

```diff
@@ -953,7 +953,10 @@ def output_detail(request, pk):
 
             #Just need the error message from one failed job
             error_msgs = dropq_compute.dropq_get_results([failed_jobs[0]], job_failure=True)
-            error_msg = error_msgs[0]
+            if error_msgs:
+                error_msg = error_msgs[0]
+            else:
+                error_msg = "Error: stack trace for this error is unavailable"
             val_err_idx = error_msg.rfind("Error")
             error = ErrorMessageTaxCalculator()
             error_contents = error_msg[val_err_idx:].replace(" ","&nbsp;")
```

Comment:
This chunk of code seems pretty suspect... Ideally I guess there would be some easily-viewable logs, and you wouldn't need to store error messages in a database. Could to comment on the logic behind this? I.e. if the error messages aren't designed to be read by users easily, what's the purpose of storing them in the database / showing them to the user instead of logging them on the server side and presenting a more generic error message?

Reply:
This error message is supposed to be a stack trace that is returned by the policy model (e.g. Tax-Calculator). When you hit previous runs that did not complete, you see the saved error message:
<img width="1283" alt="screen shot 2018-02-03 at 10 25 40 am" src="https://user-images.githubusercontent.com/9206065/35768626-c2062fb2-08cc-11e8-9dc1-5903ca78fdf2.png">

The stack trace is logged on the server side, too.

So, saving the error message in the database is more for the user than for us to introspect. The logs are a better source of info for that.